test/horreum_client_it.py

```diff
@@ -0,0 +1,101 @@
+import httpx
+import pytest
+from kiota_abstractions.authentication import BaseBearerTokenAuthenticationProvider
+from kiota_abstractions.base_request_configuration import RequestConfiguration
+from kiota_abstractions.headers_collection import HeadersCollection
+from kiota_abstractions.method import Method
+from kiota_abstractions.request_information import RequestInformation
+
+from horreum.horreum_client import new_horreum_client, HorreumClient
+from horreum.raw_client.api.test.test_request_builder import TestRequestBuilder
+from horreum.raw_client.models.protected_type_access import ProtectedType_access
+from horreum.raw_client.models.test import Test
+
+username = "user"
+password = "secret"
+
+
+@pytest.fixture()
+async def anonymous_client_without_check() -> HorreumClient:
+    print("Setting up anonymous client")
+    return await new_horreum_client(base_url="http://localhost:8080")
+
+
+@pytest.fixture()
+async def anonymous_client() -> HorreumClient:
+    print("Setting up anonymous client")
+    client = await new_horreum_client(base_url="http://localhost:8080")
```

Comment:
I believe that you can have fixtures depend on other fixtures, in which case you can remove this duplication.

Reply:
yeah they should work with nested fixtures, I think the whole test infrastructure should be optimized/refactored but I would do in ad-hoc PRs

Reply:
For further improvements https://github.com/Hyperfoil/horreum-client-python/issues/5